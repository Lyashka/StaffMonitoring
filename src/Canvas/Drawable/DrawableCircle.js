// interface CanvasInterface {
//   makeCircle() {}
//   update() {}
// }

// Rafael -> Adapter { makeCircle(...args) { Rafael.createCircle(...args)}}
//   createCircle

export class DrawableCircle {
  constructor({
    xCurrent,
    yCurrent,
    radius,
    color,
  }, two /* CanvasInterface */) {
    this.xCenter = xCurrent;
    this.yCenter = yCurrent;
    this.radius = radius;
    this.color = color;
    this.two = two;
    this.drawPoint = null;
  }

  draw() {
    this._makeCircle();
  }

  _makeCircle() {
    this.drawPoint = this.two.makeCircle(this.xCenter, this.yCenter, this.radius);
    this.drawPoint.fill = this.color;
    this.drawPoint.noStroke();
  }

  update() {
    this.two.update();
  }
}
