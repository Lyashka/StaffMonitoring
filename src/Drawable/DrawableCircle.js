export class DrawableCircle {
  constructor({ id, name, xCurrent, yCurrent, radius, color }, two) {
    this.id = id;
    this.name = name;
    this.xCenter = xCurrent;
    this.yCenter = yCurrent;
    this.radius = radius;
    this.color = color;
    this.two = two;
    this.drawPoint;
  }

  draw() {
    this._makeCircle();
    this.two.add(this.drawPoint);
    this.two.update();
  }

  _makeCircle() {
    this.drawPoint = this.two.makeCircle(this.xCenter, this.yCenter, this.radius);
    this.drawPoint.id = this.id;
    this.drawPoint.fill = this.color;
    this.drawPoint.noStroke();
  }
}
