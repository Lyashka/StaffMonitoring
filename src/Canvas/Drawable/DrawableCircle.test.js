import Two from 'two.js';
import { DrawableCircle } from './DrawableCircle';
import {
  WIDTH_CANVAS,
  HEIGHT_CANVAS,
} from 'const';

describe('Check the rendering DrawableCircle', () => {
  describe('Add a circle to canvas', () => {

    it('Draw circle with xCurrent = 550, yCurrent = 300 having radius = 40 and green color', () => {
      const canvas = document.createElement('canvas');
      const two = new Two({ width: WIDTH_CANVAS, height: HEIGHT_CANVAS, domElement: canvas });
      const circle = new DrawableCircle({
        xCurrent: 550, 
        yCurrent: 300, 
        radius: 40, 
        color: "green",
      }, two);

      circle.draw();
      circle.update();

      expect(canvas.toDataURL()).toMatchSnapshot();
    });

    it('Draw circle with xCurrent = 800, yCurrent = 200 having radius = 10 and blue color', () => {
      const canvas = document.createElement('canvas');
      const two = new Two({ width: WIDTH_CANVAS, height: HEIGHT_CANVAS, domElement: canvas });
      const circle = new DrawableCircle({
        xCurrent: 800, 
        yCurrent: 200, 
        radius: 10, 
        color: "blue",
      }, two);

      circle.draw();
      circle.update();

      expect(canvas.toDataURL()).toMatchSnapshot();
    });
  });
});
