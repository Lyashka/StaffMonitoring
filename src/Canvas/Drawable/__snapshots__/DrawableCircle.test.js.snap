// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the rendering DrawableCircle Add a circle to canvas Draw circle with xCurrent = 550, yCurrent = 300 having radius = 40 and green color 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABBUAAAJYCAYAAAA9l96PAAAABmJLR0QA/wD/AP+gvaeTAAAQJElEQVR4nO3dXaxlZ0HH4d+ejp1SYaBTUyiNoYWEgiCh8UJDMVIawRg+mmiMiaka0JBgGGJTjKnFWSXAha2NDBcqF1XxjqqEIkhsaIlYL62NgrSBqqVYCDJtp0A/HGZ70dGS05k658zHu/dZz5NMzs3JnF/eq73+WXutAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4HRZjA4AAGZi6rwWXdKyi+v//p1X/WB1zpGfVd+pHjzy8xvVPS36Usvu7szu7Jq+OaAeADgKowIAcGpMnd2in27Z66vLqld24p89ltU/V7dXt7W7W7uqR0/w/wQAtsioAACcPFM7qtdUV1a/WO0+xX/xYPWJFt3cy/t0v9D3TvHfAwC+j1EBADhxU2dVb6uuri4aVHFvi27onG5qb48PagCAWTEqAABbN3V29c7qqur8wTX/64HqhuqPmvru6BgA2M6MCgDA1lzXm1u2v7pwdMox3F/9ZlN/MToEALYrowIAsDlTF1Yfrt40OuU43VK9q6n7RocAwHZjVAAAjt/UFdVNPfkKyHVysPr1pj42OgQAthOjAgDw/9vfrg70e9Xe0Skn6CPtaa8HOQLAyWFUAACe2Qc7tyf66+onRqecJHdUb2nqwOgQAFh3RgUA4NimXlh9pvrR0Skn2b+2szd2bV8dHQIA68yoAAAc3dTLqr+tfnh0yilyX2f0ht7b3aNDAGBdGRUAgKd7fxd0qDuqF41OOcW+1s4u7dr+Y3QIAKyjHaMDAIAV88HO7VC3tv0HhaoLOtSnm9ozOgQA1pFRAQB4yv52HXko48tHp5xGP1J9sv3tGh0CAOvGqAAAPOVAv9/2ecvDZrzmyCszAYBN8EwFAOBJUz9f3Tw6Y7Cfa+qvRkcAwLowKgAANXVh9U/Vc0enDPZQO3u1BzcCwPHx9QcAoGp/BoWq53WoD42OAIB1YVQAgLmbuqJ68+iMFfLWrnMeAHA8fP0BAOZs6uzqi83j9ZGb8W/t7hVd1aOjQwBglblTAQDmbNFvZFA4mos62DtGRwDAqnOnAgDM1f52daB7qxeOTllRD1Qvbuqx0SEAsKrcqQAAc3Wgt2dQeCbnt+hXRkcAwCozKgDAHE3tqK4enbHylr0nd3YCwDEZFQBgjhb9VHXR6Iw18JKmfnJ0BACsKqMCAMzRsitHJ6wRZwUAx+B2PgCYmxt7Vgd7oHru6JQ1cbDdvcDrJQHg6dypAABz80hvyKCwGbt7pNePjgCAVWRUAIC5WbpA3rRll41OAIBVZFQAgPlxgbx5l48OAIBV5JkKADAnU+dVX89ngM06XD2/qf8aHQIAq8SdCgAwJ4suyaCwFTuqV4+OAIBVY1QAgDlZdvHohLW1cHYAsJFRAQDmxYXxVhlkAOBpjAoAMC8vHR2wxowKALCBUQEA5uUFowPWmLMDgA2MCgAwL88ZHbDGnB0AbGBUAIB5cWG8dc4OADYwKgDAvDx7dMAaMyoAwAZGBQAAAGBLjAoAMC/fHh2wxh4ZHQAAq8aoAADz4sJ465wdAGxgVACAeXFhvHXODgA2MCoAwLx8fXTAGnN2ALCBUQEA5uWe0QFr7O7RAQCwaowKADAvLoy3auHsAGAjowIAzIkL461bOjsA2MioAABz8gP9Y7UcnbGGDndmd46OAIBVY1QAgDm5pm9W/zI6Yw3d1TV9a3QEAKwaowIAzM9towPWkDMDgKMwKgDA/LhA3jxnBgBHYVQAgLnZ3a3Vw6Mz1sjD7e720REAsIqMCgAwN1f1aPWXozPWyMeOnBkAsIFRAQDm6c9HB6yNHc4KAI7FqAAA8/R31b2jI9bAl/vd/n50BACsKqMCAMzR1OEW3TA6Yw1cXy1HRwDAqjIqAMBcndNN1X+Ozlhh97enPxsdAQCrzKgAAHO1t8dbdOPojBV2fXt7fHQEAKwyowIAzNmyP6z+fXTGCvpK9ZHREQCw6owKADBnU99t0TtHZ6ycRe9u6rHRGQCw6owKADB3+/qb6pbRGSvk4+3rU6MjAGAdGBUAgKp3VQ+NjlgBD7azd4+OAIB1YVQAAGrqvurK5v36xGWL3t61fXV0CACsizNGBwAAK+Jz3dPr2lP9+OiUQf6gqQ+NjgCAdeJOBQDgKXv6reofRmcM8Pnqt0dHAMC6MSoAAE/Z2+Od1c9Wd41OOY2+UF3R1BOjQwBg3SxGBwAAK+j9XdCh7qheNDrlFLu/uvTIMyUAgE1ypwIA8HTX9rXO6I21rS+276suNygAwNa5UwEAOLYPdH7/3WeqV41OOcm+2M5+xpseAODEuFMBADi23+mB6rK218MbP9+uXmtQAIAT55WSAMAz+1yP9ro+Wu2qLm1973RcVh+uruy9fWd0DABsB+v6oQAAGOG63tqyP6nOGZ2ySQdb9Gvt6+bRIQCwnfj6AwBw/Pb1iXZ2SXXL6JRN+Hg7e6VBAQBOPncqAABbM/Wman910eiUY7i3RXvb16dGhwDAdmVUAAC27sae1cHeUV1dXTA654j7W3RDy/64qcdGxwDAdmZUAABO3P529WC/2rL3VC8ZVPHl6vrqT5t6YlADAMyKUQEAOLne1491uF+ufqk69xT/tYd78vkOH23qsz35hgcA4DQxKgAAp8bUWS26vGWXV5dVr+rEHxJ9uLqrur36bHWbrzgAwDhGBQDg9Jj6oRZdUr20ZS+rLq7Oq55TPa969pHf/Hb1UPVI9Y3qnhZ9qWV3d2Z3dk3fGlAPAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADDU/wAjKiQ9VvJu4gAAAABJRU5ErkJggg=="`;

exports[`Check the rendering DrawableCircle Add a circle to canvas Draw circle with xCurrent = 800, yCurrent = 200 having radius = 10 and blue color 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABBUAAAJYCAYAAAA9l96PAAAABmJLR0QA/wD/AP+gvaeTAAAKrElEQVR4nO3cMU4bURQF0AsIV8lqyD5AEMKWQmAZUepkH0kNEpAyG4A6L4XHKWiw8Bu+BedIrxmNr95vrzw/AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABmVYukzpL6ltR1Ug/TXE/PPi3fAQAAAPivTpK6S6qemNukjkdvCwAAAAxXe0ldrlEmPJ6LpHZHbw8AAAAM86xCYTXno7cHAAAAhqiTDQqF1RyNPgUAAADwomox3Y+waanw2+WNAAAA8KbUWUOhsJrT0acB1uMiFAAAoEPnZwuHjVnAjJQKAABAh4PGrA+NWcCMdkYvAAAAvAZ1n+RdU9hDsvO+KQuYkX8qAAAA2+bv6AWA9SgVAACADn+2NAuYkVIBAADo8Ksx62djFjAjpQIAANDhe2PWj8YsAAAAYLvVflI3SdWGc5fUYvRpAAAAgBdVxw2lwuHoUwAAAABD1MUGhcLn0dsDAAAAw9RuUl+eUSicL38LAAAAvHF1tOYdCzc+eQAAAAAeqf2kTpP6mtRVUvfTXE3PPi7fAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAXqd/VUmEOEz18IkAAAAASUVORK5CYII="`;
